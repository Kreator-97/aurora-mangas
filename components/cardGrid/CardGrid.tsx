import { FC } from 'react'
import { Serie } from '../../interfaces'

import { CardSuscribe } from '../'

interface Props {
  series    ?: Serie[];
  children  ?: React.ReactNode;
  gridCols ?: string;
}

export const CardGrid: FC<Props> = ({ series, children, gridCols }) => {
  let gridDefault = 'grid-cols-1 md:grid-cols-2'

  if( gridCols ) {
    gridDefault = gridCols
  }

  return (
    <div className={ `grid gap-2 ${gridDefault}` }>
      {
        series?.map((serie) =>(
          <CardSuscribe
            key={serie.name}
            serie={serie}
          />
        ))
      }
      { children }
    </div>
  )
}
