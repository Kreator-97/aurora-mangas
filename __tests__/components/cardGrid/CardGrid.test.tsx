import { render, screen } from '@testing-library/react'

import { CardGrid } from '../../../components/cardGrid/CardGrid'
import { series } from '../../fixtures/db'

describe('tests on CardGrid component', () => {
  test('should to match with snapshot', () => {
    // we render an array with 2 series
    const component = render(<CardGrid series={[series[0], series[1]]} />)
    const cardGrid = screen.getByTestId('card-grid')
    
    expect(component.container).toMatchSnapshot()
    expect(cardGrid).toBeInTheDocument
    expect(screen.getAllByTestId('card-serie').length ).toBe(2)
  })

})

