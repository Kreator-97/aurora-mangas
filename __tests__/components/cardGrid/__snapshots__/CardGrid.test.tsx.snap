// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on CardGrid component should to match with snapshot 1`] = `
<div>
  <div
    class="grid gap-2 grid-cols-1 md:grid-cols-2 px-2"
    data-testid="card-grid"
  >
    <div
      class="relative w-full max-w-[var(--md)]"
      data-testid="card-serie"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 50% 0px 0px 0px;"
        />
        <img
          alt="Inuyasha"
          class="object-cover"
          data-nimg="responsive"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div
        class="absolute bottom-0 bg-dark bg-opacity-75 w-full p-2 flex items-center justify-between "
      >
        <div>
          <h2
            class="text-lg sm:text-xl"
          >
            Serie: 
            Inuyasha
          </h2>
          <p
            class="font-light"
          >
            Periodicidad: 
            MENSUAL
          </p>
        </div>
        <button
          class="btn bg-accent"
        >
          Suscribirse
        </button>
      </div>
    </div>
    <div
      class="relative w-full max-w-[var(--md)]"
      data-testid="card-serie"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 50% 0px 0px 0px;"
        />
        <img
          alt="One Piece"
          class="object-cover"
          data-nimg="responsive"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div
        class="absolute bottom-0 bg-dark bg-opacity-75 w-full p-2 flex items-center justify-between "
      >
        <div>
          <h2
            class="text-lg sm:text-xl"
          >
            Serie: 
            One Piece
          </h2>
          <p
            class="font-light"
          >
            Periodicidad: 
            MENSUAL
          </p>
        </div>
        <button
          class="btn bg-accent"
        >
          Suscribirse
        </button>
      </div>
    </div>
  </div>
</div>
`;
